src/jaxsim/rbda/contacts/visco_elastic.py

```diff
@@ -0,0 +1,1050 @@
+            return False
+
+        return hash(self) == hash(other)
+
+
+@jax_dataclasses.pytree_dataclass
+class ViscoElasticContacts(common.ContactModel):
+    """Visco-elastic contacts model."""
+
+    parameters: ViscoElasticContactsParams = dataclasses.field(
+        default_factory=ViscoElasticContactsParams
+    )
+
+    terrain: jax_dataclasses.Static[Terrain] = dataclasses.field(
+        default_factory=FlatTerrain
+    )
+
+    max_squarings: jax_dataclasses.Static[int] = 25
+
+    @classmethod
+    def build(
+        cls: type[Self],
+        parameters: SoftContactsParams | None = None,
+        terrain: Terrain | None = None,
+        model: js.model.JaxSimModel | None = None,
+        max_squarings: jtp.IntLike | None = None,
+        **kwargs,
+    ) -> Self:
+        """
+        Create a `ViscoElasticContacts` instance with specified parameters.
+
+        Args:
+            parameters: The parameters of the soft contacts model.
+            terrain: The considered terrain.
+            model:
+                The robot model considered by the contact model.
+                If passed, it is used to estimate good default parameters.
+            max_squarings:
+                The maximum number of squarings performed in the matrix exponential.
+
+        Returns:
+            The `ViscoElasticContacts` instance.
+        """
+
+        if len(kwargs) != 0:
+            logging.debug(msg=f"Ignoring extra arguments: {kwargs}")
+
+        # Build the contact parameters if not provided. Use the model to estimate
+        # good default parameters, if passed. Users can later override these default
+        # parameters with their own values -- possibly tuned better.
+        if parameters is None:
+            parameters = (
+                ViscoElasticContactsParams.build_default_from_jaxsim_model(model=model)
+                if model is not None
+                else cls.__dataclass_fields__["parameters"].default_factory()
+            )
+
+        return ViscoElasticContacts(
+            parameters=parameters,
+            terrain=terrain or cls.__dataclass_fields__["terrain"].default_factory(),
+            max_squarings=int(
+                max_squarings or cls.__dataclass_fields__["max_squarings"].default()
+            ),
+        )
+
+    @classmethod
+    def zero_state_variables(cls, model: js.model.JaxSimModel) -> dict[str, jtp.Array]:
+        """
+        Build zero state variables of the contact model.
+        """
+
+        # Initialize the material deformation to zero.
+        tangential_deformation = jnp.zeros(
+            shape=(len(model.kin_dyn_parameters.contact_parameters.body), 3),
+            dtype=float,
+        )
+
+        return {"tangential_deformation": tangential_deformation}
+
+    @jax.jit
+    def compute_contact_forces(
+        self,
+        model: js.model.JaxSimModel,
+        data: js.data.JaxSimModelData,
+        *,
+        dt: jtp.FloatLike,
```

Comment:
It is never set because this contact model needs to know what is the integration step so that it can compute the average contact forces by integrating them in continuous time. You can think of this `dt` in a similar way of the requirement of a semi-implicit integrator.

For this reason, `dt` is a necessary parameter that must be provided by the caller. Setting a default, in my opinion, can result to unpleasant surprises. In the future, if we manage to store this `dt` inside the model as we are already experimenting, we can make this optional.